### PR TITLE
Lazy loaded Block Extension

### LINKED ISSUE #675: Make BlockExtension a lazy extension
## Feature Request

The `BlockExtension` should be [lazy loaded](https://symfony.com/doc/current/templating/twig_extension.html#creating-lazy-loaded-twig-extensions) (dependency-less).

### PR DESCRIPTION
**Pull request by mikemix**
## Subject

I am targeting this branch, because it's a BC feature which makes the BlockExtension [lazy](https://symfony.com/doc/current/templating/twig_extension.html#creating-lazy-loaded-twig-extensions).

Closes #675 

## Changelog

```markdown
### Changed
* Lazy `BlockExtension` to speed Twig initialization
```

### CODE REVIEW

**Comment by mikemix** [comment]
I'm not sure where's that error coming from ;)

File: src/Resources/config/core.xml [hunk 1]
```diff
@@ -34,6 +35,7 @@
             <argument type="service" id="sonata.block.cache.handler.default" on-invalid="ignore"/>
             <argument type="service" id="debug.stopwatch" on-invalid="ignore"/>
         </service>
+        <service id="Sonata\BlockBundle\Templating\Helper\BlockHelper" alias="sonata.block.templating.helper"/>
```
Comment (mikemix) [inline src/Resources/config/core.xml]: Required, because `TwigFunction` expects a callable to begin with.

**Comment by VincentLanglet** [comment]
HI @mikemix ! What's the status of this ? Do you have time to finish your PR ? :)

**Comment by mikemix** [comment]
It's ready since 30th of January.

**Comment by VincentLanglet** [comment]
There is the phpstan build failed, so this can't be merge.
Maybe a rebase can fix the issue...

**Comment by mikemix** [comment]
I don't know what's causing this tbh. Event dispatcher error?

**Comment by VincentLanglet** [comment]
> I don't know what's causing this tbh. Event dispatcher error?

You need to rebase the 4.x branch because I think it was fixed. @mikemix

**Comment by mikemix** [comment]
I'll do that. Also I'm going to fix the test. Late at night this evening probably.

**Comment by mikemix** [comment]
This is still WIP.

**Comment by mikemix** [comment]
@VincentLanglet I don't like these tests ;) a real test using the `IntegrationTestCase` would be much better:

https://twig.symfony.com/doc/2.x/advanced.html#testing-an-extension

File: tests/Twig/Extension/BlockExtensionTest.php [hunk 2]
```diff
@@ -41,17 +42,23 @@ public function setUp(): void
...
-        $loader = $this->createMock(LoaderInterface::class);
+        $this->blockExtension = new BlockExtension();
 
-        $this->blockExtension = new BlockExtension($this->blockHelper);
-
-        $this->env = new Environment($loader);
+        $this->env = new Environment($this->createMock(LoaderInterface::class));
         $this->env->addExtension($this->blockExtension);
+        $this->env->addRuntimeLoader(new FactoryRuntimeLoader([
+            BlockHelper::class => function () {
```
Comment (greg0ire) [inline tests/Twig/Extension/BlockExtensionTest.php]:
```suggestion
            BlockHelper::class => function (): BlockHelper {
```

**Comment by mikemix** [comment]
We will save tons of CO2 emission with this one I'm sure.

**Comment by greg0ire** [comment]
Thanks on fighting climate change!

**Pull request merged**